packages/core/src/defer/interfaces.ts

```diff
@@ -137,6 +169,29 @@ export enum DeferBlockState {
   Error = 3,
 }
 
+/**
+ * Represents defer trigger types.
+ */
+export enum DeferBlockTrigger {
```

Comment:
nit: doing `const enum` vs `enum` to inline values (vs having the need to retain the whole enum/object):
```suggestion
export const enum DeferBlockTrigger {
```

Reply:
const enums cannot be iterated over, and we do that specifically in the `incremental.ts` file. So not retaining it ends up being a problem.

Reply:
OK addressed